src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -63,14 +72,46 @@ def _process_model_before_weight_loading(self, model, **kwargs):
         from compressed_tensors.quantization import apply_quantization_config
 
         ct_quantization_config = self.compressor.quantization_config
-        apply_quantization_config(model, ct_quantization_config, run_compressed=True)
+
+        if self.run_compressed and self.is_compressed:
+            apply_quantization_config(model, ct_quantization_config, run_compressed=True)
 
     def _process_model_after_weight_loading(self, model, **kwargs):
-        pass
+        """Decompress loaded model if necessary - need for qat"""
+
+        if not self.run_compressed or not self.is_compressed:
+            config = kwargs.get("config", None)
+            cache_path = config._name_or_path
+            if not os.path.exists(cache_path):
+                from huggingface_hub import hf_hub_download
+
+                from transformers import TRANSFORMERS_CACHE
+                from transformers.utils import http_user_agent
+
+                user_agent = http_user_agent()
+                config_file_path = hf_hub_download(
+                    repo_id=cache_path,
+                    filename="config.json",
+                    cache_dir=TRANSFORMERS_CACHE,
+                    force_download=False,
+                    user_agent=user_agent,
+                )
+                cache_path = os.path.sep.join(config_file_path.split(os.path.sep)[:-1])
+
+            from compressed_tensors.quantization import QuantizationStatus
+
+            self.compressor.quantization_config.quantization_status = QuantizationStatus.FROZEN
+
+            self.compressor.decompress(model_path=cache_path, model=model)
```

Comment:
The `model_path` should point to the compressed model on disk no? @horheynm

Reply:
Yeah that will be disk if from_pretained(...) was provided a disk path. Otherwise disk will be the hf cache
Config in there is the path provided to the pre trained